src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1092,13 +1086,25 @@ private void WriteApphostsForShimRuntimeIdentifiers()
             /// </summary>
             private bool CanResolveApphostFromFrameworkReference()
             {
-                if (_targetFramework.Version.Major >= 3
-                    && _targetFramework.Framework.Equals(".NETCoreApp", StringComparison.OrdinalIgnoreCase))
+                if (!CanWriteToCacheFile)
```

Comment:
any test is covering these code?

Reply:
the cache file cannot be written is a new thing?

Reply:
Yes, I made this code change in response to test failures in `DesignTimeBuildTests`.

The scenario is when we are doing a design-time build after there has been a change to the project file.  In that case, the assets file doesn't necessarily match the information in the project file.  In that case we shouldn't save the result of ResolvePackageAssets to the cache.  This logic has been in there for a while, it just needed updating with the TargetFramework changes.